Microsoft.Toolkit.Uwp.UI/Extensions/ListViewBase/SmoothScrollIntoView.cs

```diff
@@ -0,0 +1,198 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+
+namespace Microsoft.Toolkit.Uwp.UI.Extensions
+{
+    /// <summary>
+    /// Smooth scroll the list to bring specified item into view
+    /// </summary>
+    public static class SmoothScrollIntoView
+    {
+        /// <summary>
+        /// Smooth scrolling the list to bring the specified index into view
+        /// </summary>
+        /// <param name="listViewBase">List to scroll</param>
+        /// <param name="index">The intex to bring into view</param>
+        /// <param name="itemPlacement">Set the item placement after scrolling</param>
+        /// <param name="disableAnimation">Set true to disable animation</param>
+        /// <param name="scrollIfVisibile">Set true to disable scrolling when the corresponding item is in view</param>
+        /// <param name="additionalHorizontalOffset">Adds additional horizontal offset</param>
+        /// <param name="additionalVerticalOffset">Adds additional vertical offset</param>
+        /// <returns>Note: Even though this return <see cref="Task"/>, it will not wait until the scrolling completes</returns>
+        public static async Task SmoothScrollIntoViewWithIndex(this ListViewBase listViewBase, int index, ItemPlacement itemPlacement = ItemPlacement.Default, bool disableAnimation = false, bool scrollIfVisibile = true, int additionalHorizontalOffset = 0, int additionalVerticalOffset = 0)
+        {
+            if (index > (listViewBase.Items.Count - 1))
+            {
+                index = listViewBase.Items.Count - 1;
+            }
+
+            index = (index < 0) ? (index + listViewBase.Items.Count) : index;
+
+            bool isVirtualizing = default;
+            double previousXOffset = default, previousYOffset = default;
+
+            var scrollViewer = listViewBase.FindDescendant<ScrollViewer>();
+            var selectorItem = listViewBase.ContainerFromIndex(index) as SelectorItem;
+
+            if (selectorItem == null)
+            {
+                isVirtualizing = true;
+
+                previousXOffset = scrollViewer.HorizontalOffset;
+                previousYOffset = scrollViewer.VerticalOffset;
+
+                var tcs = new TaskCompletionSource<object>();
+
+                void ViewChanged(object obj, ScrollViewerViewChangedEventArgs args) => tcs.TrySetResult(result: null);
+
+                try
+                {
+                    scrollViewer.ViewChanged += ViewChanged;
+                    listViewBase.ScrollIntoView(listViewBase.Items[index], ScrollIntoViewAlignment.Leading);
```

Comment:
This is changing the position of the list view. Isn't it breaking the logic below ?

Reply:
No, because I am scrolling it back

https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/b57e6df24ca205561d12e815552496a60378db61/Microsoft.Toolkit.Uwp.UI/Extensions/ListViewBase/SmoothScrollIntoView.cs#L79-L83